pkgs/top-level/python-packages.nix

```diff
@@ -91,11 +91,9 @@ let
 
   # Get list of required Python modules given a list of derivations.
   requiredPythonModules = drvs: let
-    filterNull = list: filter (x: !isNull x) list;
-    conditionalGetRecurse = attr: condition: drv: let f = conditionalGetRecurse attr condition; in
-      (if (condition drv) then unique [drv]++(concatMap f (filterNull(getAttr attr drv))) else []);
-    _required = drv: conditionalGetRecurse "propagatedBuildInputs" hasPythonModule drv;
-  in [python] ++ (unique (concatMap _required (filterNull drvs)));
+    removeNull = list: filter (x: !isNull x) list;
+    modules = filter hasPythonModule (removeNull drvs);
+  in unique [python] ++ (modules ++ (flatten (catAttrs "requiredPythonModules" modules)));
```

Comment:
This is parsed as `(unique [python]) ++ (...)`, not what you want.

`unique` is expensive (O(n²)), may it be removed?

Reply:
`concatLists` (which is a builtin reexported from lib) may be used instead of `flatten`.

Reply:
In principle it can be removed. This is currently only evaluated when using `python.buildEnv`.